solutions/cf-616D.mdx

```diff
@@ -0,0 +1,79 @@
+---
+id: cf-616D
+source: CF
+title: Longest k-Good Segment
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/22712)
+
+<LanguageSection>
+
+<JavaSection>
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+## Java Implementation
+
+```java
+public class kgood {
+	CodeSnip{Kattio}
+
+    public static int MAX = (int) 1e6;
+
+    public static void main(String[] args) throws IOException {
+        Kattio io = new Kattio();
+        int N = io.nextInt();
+        int K = io.nextInt();
+
+        int[] A = new int[MAX];
+        int[] freq = new int[MAX]; // Frequency of each number in the window.
```

Comment:
1. again, replace `MAX` with `N`
2. i think the style is 2 lines after a line of code for a comment (like `int x = 1;  // this is a comment`)